Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp

```diff
@@ -4208,72 +4215,24 @@ static void ensureReferenceAndAddWatchpoint(VM&, PolymorphicAccessJITStubRoutine
 {
-    SuperSamplerScope superSamplerScope(false);
-
-    dataLogLnIf(InlineCacheCompilerInternal::verbose, "Regenerate with m_list: ", listDump(poly.m_list));
-
-    // Regenerating is our opportunity to figure out what our list of cases should look like. We
-    // do this here. The newly produced 'cases' list may be smaller than m_list. We don't edit
-    // m_list in-place because we may still fail, in which case we want the PolymorphicAccess object
-    // to be unmutated. For sure, we want it to hang onto any data structures that may be referenced
-    // from the code of the current stub (aka previous).
-    Vector<WatchpointSet*, 8> additionalWatchpointSets;
-    Vector<Ref<AccessCase>, 16> cases;
-    cases.reserveInitialCapacity(poly.m_list.size());
-    unsigned srcIndex = 0;
-    for (auto& someCase : poly.m_list) {
-        [&] () {
-            if (!someCase->couldStillSucceed())
-                return;
-
-            auto sets = collectAdditionalWatchpoints(vm(), someCase.get());
-            for (auto* set : sets) {
-                if (!set->isStillValid())
-                    return;
-            }
-
-            // Figure out if this is replaced by any later case. Given two cases A and B where A
-            // comes first in the case list, we know that A would have triggered first if we had
-            // generated the cases in a cascade. That's why this loop asks B->canReplace(A) but not
-            // A->canReplace(B). If A->canReplace(B) was true then A would never have requested
-            // repatching in cases where Repatch.cpp would have then gone on to generate B. If that
-            // did happen by some fluke, then we'd just miss the redundancy here, which wouldn't be
-            // incorrect - just slow. However, if A's checks failed and Repatch.cpp concluded that
-            // this new condition could be handled by B and B->canReplace(A), then this says that we
-            // don't need A anymore.
-            //
-            // If we can generate a binary switch, then A->canReplace(B) == B->canReplace(A). So,
-            // it doesn't matter that we only do the check in one direction.
-            for (unsigned j = srcIndex + 1; j < poly.m_list.size(); ++j) {
-                if (poly.m_list[j]->canReplace(someCase.get()))
-                    return;
-            }
-
-            // If the case had been generated, then we have to keep the original in m_list in case we
-            // fail to regenerate. That case may have data structures that are used by the code that it
-            // had generated. If the case had not been generated, then we want to remove it from m_list.
-            cases.append(someCase);
-
-            additionalWatchpointSets.appendVector(sets);
-        }();
-        ++srcIndex;
+    // Accidentally, it already includes megamorphic case. Then we just return it.
```

Comment:
How does this happen? Or do you mean incidentally?

Reply:
Sometimes, after putting this metamorphic access case, we fail to generate code and leave them in PolyList.